Rewrite of Script in Python with more configuration options

ghan1t opened the pull request:
I rewrote most of the script to work only with python (except the trigger.sh) with more config options (zfs-autobackup parameters), install options, logging and email on success or error. Tested on my TrueNAS Scale.

You would need to change REPO_URL in trigger.sh to your own repo after merging this PR.

ghan1t committed:
work in progress for replacement ob backup.sh with a python script. Added more flexible configuration for autobackup parameters, logging and smtp/mail.

ghan1t committed:
Added python backup script calling zfs-autobackup from python.
Added autobackup parameters to config.
Extended trigger.sh to install this script and dependencies from github.

ghan1t committed:
Deleted old config file
Updated .gitignore to exclude __pycache__/ venv/ and config.yaml

ghan1t committed:
fixing errors in trigger.sh

ghan1t committed:
Improving logging and Readme.
Deleted old backup.sh and monitor.py

ghan1t committed:
renamed monitor2.py back to monitor.py

ghan1t committed:
Updated readme

ghan1t committed:
Fixed error in monitor and readme.

ghan1t committed:
Fixed mistake when checking out newest tag if no tags exist.
Simplified checkout code.

ghan1t committed:
git clone does not work when downloading trigger.sh with curl.
Changing to git init and then git remode add.

ghan1t committed:
Added check-monitor option
Added description for failing PyYML
Fixed error in import of zfs-autobackup

Jip-Hop commented:
Nice to see you've picked up development of udev-trigger-zfs-autobackup. As I'm currently not using this myself and not maintaining this repo I won't merge this. Instead I hope others will find your fork by leaving this pull request open.

ghan1t committed:
added --test flag to test a backup through the python script without importing/exporting the pool.
Removed 'set -eu' because they hid errors.

ghan1t committed:
Added opption -f to delete a broken venv before reinstalling.

ghan1t committed:
Capturing std:err of zfs-autobackup and report it in mail and log output.
This might solve an issue where zfs-autobackup threw an error when writing to std:err.
Also removed redundant zfs_tag from config.

ghan1t committed:
Added parameter to split zfs-autobackup parameters if they contain a space. Spaces are not allowed unless escaped and mess up the command line parameters. It's true by default but can be turned off. This fixes #1

ghan1t committed:
Fixing mistake with checking empty std:err of zfs-autobackup

ghan1t committed:
Fixing bug with wrong pool_config as parameter. Fixes #3

ghan1t committed:
Adding missing AppConfig to method decrypt_and_backup

ghan1t committed:
Fixing missing captured_output in logging and improved exception logging.

ghan1t committed:
Updating dependencies

ghan1t committed:
Fixing bug when smtp config is left out. Fixing #6

ghan1t committed:
Merge pull request #10 from Jip-Hop/main

Create LICENSE
